packages/opentelemetry-core/src/trace/TracerDelegate.ts

```diff
@@ -24,62 +24,77 @@ import { NoopTracer } from './NoopTracer';
 // to be changed/disabled during runtime without needing to change reference
 // to the global tracer
 export class TracerDelegate implements types.Tracer {
-  private readonly _userTracer: types.Tracer | null;
-  private readonly _fallbackTracer: types.Tracer;
-  private _tracer: types.Tracer;
+  private _currentTracer: types.Tracer;
 
   // Wrap a tracer with a TracerDelegate. Provided tracer becomes the default
   // fallback tracer for when a global tracer has not been initialized
-  constructor(tracer?: types.Tracer | null, fallbackTracer?: types.Tracer) {
-    this._userTracer = tracer || null;
-    this._fallbackTracer = fallbackTracer || new NoopTracer();
-    this._tracer = this._userTracer || this._fallbackTracer; // equivalent to this.start()
+  constructor(
+    private readonly tracer: types.Tracer | null = null,
+    private readonly fallbackTracer: types.Tracer = new NoopTracer()
+  ) {
+    this._currentTracer = tracer || fallbackTracer; // equivalent to this.start()
   }
 
   // Begin using the user provided tracer. Stop always falling back to fallback tracer
   start(): void {
-    this._tracer = this._userTracer || this._fallbackTracer;
+    this._currentTracer = this.tracer || this.fallbackTracer;
```

Comment:
Do we need to call `.start()` on `_currentTracer` if it has not been called yet? And do we need to track whether `this.tracer` and `this.fallbackTracer` have been started yet?

Reply:
The constructor should handle the initial call to `.start()`. Note: typescript will complain about `_currentTracer` not being initialized if I replaced it with an actual call to `this.start()`.

https://github.com/open-telemetry/opentelemetry-js/blob/819227eca6d5f718a60a73b8edc36e080ae93fd1/packages/opentelemetry-core/src/trace/TracerDelegate.ts#L31-L36

>And do we need to track whether this.tracer and this.fallbackTracer have been started yet?

Are you proposing adding an `_isStarted` bool to `TracerDelegate`?

Reply:
So I guess my thinking is that if we later decide to pass config options to the `.start` method that we will need this `TracerDelegate` to pass that config on to the tracer.  Here's a simplified possible scenario:

```typescript
// basic-tracer/index.ts
class BasicTracer {
  start(config: Config) {
     ...
  }
   ...
}
export const tracer = new TracerDelegate(new BasicTracer(), new NoopTracer());

// customer code
import {tracer} from '@opentelemetry/basic-tracer';
tracer.start({exporter: new VendorExporter()}); // This config should get passed into `BasicTracer.start` via `TracerDelegate`
```

So in that case, with `TracerDelegate` as written, the config wouldn't be passed into `BasicTracer.start`.

For now, the `start` function takes no arguments, so I think this can be a refactoring in the future. 

Overall, I think we should merge this PR and then improve later on.